src/openrct2/ride/Ride.cpp

```diff
@@ -7399,6 +7399,11 @@ bool ride_has_adjacent_station(Ride* ride)
 bool ride_has_station_shelter(Ride* ride)
 {
     auto stationObj = ride_get_station_object(ride);
+    if (network_get_mode() != NETWORK_MODE_NONE)
+    {
+        // The server might run in headless mode so no images will be loaded, only check for stations.
+        return stationObj != nullptr;
```

Comment:
This means that plain stations will be regarded as sheltered in network play. After all, this can be the only reason it even checks for `BaseImageId`. I would prefer if it could check for the number of images in the object, or if that isn't possible, if we could add a flag to station objects that determines this status.

Reply:
In headless mode there are no image information at all. But yeah I would think its better to flag specific objects as sheltered since such metadata shouldn't rely on the image.